builder/googlecompute/config.go

```diff
@@ -494,6 +497,10 @@ func (c *Config) Prepare(raws ...interface{}) ([]string, error) {
 		c.DiskName = c.InstanceName
 	}
 
+	if c.ImageSourceDisk == "" {
```

Comment:
I think the base idea is there, but there remains some questions to be answered:

1. Can we be sure the disk we pick here exists? I think in prepare we should check that it does, and if it doesn't, we should error immediately.
2. This option makes sense for disks that are not the one we base the instance on, what if they're marked auto-delete? We should probably error if we are relying on one that gets deleted, or double-check at least that it doesn't prevent from creating the image at the end (note: looking at the code I would presume it does as the disks get created when the instance is shutdown).

Reply:
I had the same idea, but I wanted to keep the PR simple to gauge a reaction before I invested more time in fleshing out the error conditions - so yeah, sounds good! I'll look into that.

And good point about auto-deleting disks, I'll look into that as well.